config/ingest.go

```diff
@@ -41,9 +41,9 @@ type Ingest struct {
 	// IngestWorkerCount sets how many ingest worker goroutines to spawn. This
 	// controls how many concurrent ingest from different providers we can handle.
 	IngestWorkerCount int
-	// KeepAdvertisements, when true, prevents advertisements from being
-	// deleted after they are ingested.
-	KeepAdvertisements bool
+	// KeepAdvertisementsCarDir specifies a directory to write CAR files
+	// containing advertisements that have been ingested.
+	KeepAdvertisementsCarDir string
```

Comment:
- Should this be changed to `CarSaveLocation` or `MirrorLocation` and take the form of a URI?
- Do we want to save to multiple locations, like directory and S3?
- Do we want the indexer to be responsible for moving the CAR files to the final mirror location (e.g. S3) or should a separate process do that?

Reply:
Ideally, we want to _stream_ the car files directly to S3. That may involve temporary storage of files on local disk. 

Instead of string location, the config should allow "CarWriter" config or something along those lines with 2 implementations: one to write on local disk that asks for a destination path, and one that uploads to S3 directly.

Please leave the implementation of the latter writer out for a separate PR.

Reply:
Streaming directly to S3 has two disadvantages:
1. Requires a separate detached index for the CAR file.
2. Cannot use a separate process to send files to S3 or another destination.

If we care about either of these then it may be worth writing CAR files to the local filesystem temporarily.

Reply:
I'm less concerned about S3 specifically as the long-term goal. If we can get to car files in local memory/disk so that we can turn them into filecoin deals or similar that should be just about as good

Reply:
> Requires a separate detached index for the CAR file.

Fine to temporarily write them to disk. This should be encapsulated in the exporter implementation. Even without it, the deal making and upload to FileCoin without a doubt needs to be a separate service/thing from storetheindex and it is easy enough to generate index there if needed, assuming we cannot upload carv1s straight up.

> Cannot use a separate process to send files to S3 or another destination.

Off the top of my head we can have a separate process or a side car container that watches a directory and uploads. What am I missing?

Reply:
Missing nothing. We can still export CARs to local filesystem and then write to S3 using a separate process if we want.